### ISSUE TITLE
Revamp installation issue

### ISSUE THREAD

**Issue by zhouyuan1119** [cornell-zhang/datuner#25]
When installing the revamp branch, if the user already has `python` and `pip` in his/her `PATH`, the installation script will error out with the following error message:

```
--2018-04-05 13:21:29--  https://bootstrap.pypa.io/get-pip.py
Resolving bootstrap.pypa.io (bootstrap.pypa.io)... 151.101.116.175
Connecting to bootstrap.pypa.io (bootstrap.pypa.io)|151.101.116.175|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1780465 (1.7M) [text/x-python]
Saving to: ‘get-pip.py’

100%[===============================================================================================>] 1,780,465   --.-K/s   in 0.1s    

2018-04-05 13:21:29 (12.5 MB/s) - ‘get-pip.py’ saved [1780465/1780465]

/tmp/tmpHUQnQQ/pip.zip/pip/_vendor/urllib3/util/ssl_.py:339: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
/tmp/tmpHUQnQQ/pip.zip/pip/_vendor/urllib3/util/ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
Requirement already up-to-date: pip in /work/zhang-x1/users/yz882/cornell-zhang/datuner/build/pkgs/python/install/lib/python2.7/site-packages
/tmp/tmpHUQnQQ/pip.zip/pip/_vendor/urllib3/util/ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
cd /work/zhang-x1/users/yz882/test/datuner-revamp/datuner/build/pkgs/python/install/bin; \
export LD_LIBRARY_PATH=/work/zhang-x1/users/yz882/test/datuner-revamp/datuner/build/pkgs/python/install/lib; \
./pip install opentuner;
/bin/sh: line 2: ./pip: No such file or directory
```
The current installation script only works on a clean environment. I believe the master branch also has this problem.

**Comment by Hecmay** [comment]
I cannot reproduce the error. The output of my installing DATuner revamp on `zhang-05`:
```
cd /work/zhang-x1/users/sx233/software/datuner/build/pkgs/python/install/bin; \
        export LD_LIBRARY_PATH=/work/zhang-x1/users/sx233/software/datuner/build/pkgs/python/install/lib; \
        ./pip install opentuner;
Collecting opentuner
/work/zhang-x1/users/sx233/software/datuner/build/pkgs/python/install/lib/python2.7/site-packages/pip/_vendor/urllib3/util/ssl_.py:339: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  SNIMissingWarning
/work/zhang-x1/users/sx233/software/datuner/build/pkgs/python/install/lib/python2.7/site-packages/pip/_vendor/urllib3/util/ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecurePlatformWarning
  Downloading opentuner-0.8.0.tar.gz (64kB)
```
The `InsecurePlatformWarning` and `SNIMissingWarning` are just common warning messages for Python with version < 2.7.9 and will not interrupt the installation. 

It seems that the error was thrown out when executing `./python get-pip.py; ` in line 45 of `DATUNER_HOME/pkgs/python-opentuner/Makefile`
```bash
$(PythonInstallDir)/bin/pip:
        cd $(PythonInstallDir)/bin; \
        export LD_LIBRARY_PATH=$(PythonInstallDir)/lib; \
        wget https://bootstrap.pypa.io/get-pip.py; \
        ./python get-pip.py; \
```
the `./python` that is used to run `get-pip.py` is the one in `$(PythonInstallDir)/bin`, so I do not think it is caused by the the disturbance of the environment `PATH`.

**Comment by Hecmay** [comment]
It seems that `pip` is already installed in `site-packages`, but the bash cannot find the binary `pip` under the `pkgs/bin` path.
```
Requirement already up-to-date: pip in /work/zhang-x1/users/yz882/cornell-zhang/datuner/build/pkgs/python/install/lib/python2.7/site-packages
/tmp/tmpHUQnQQ/pip.zip/pip/_vendor/urllib3/util/ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
cd /work/zhang-x1/users/yz882/test/datuner-revamp/datuner/build/pkgs/python/install/bin; \
export LD_LIBRARY_PATH=/work/zhang-x1/users/yz882/test/datuner-revamp/datuner/build/pkgs/python/install/lib; \
./pip install opentuner;
/bin/sh: line 2: ./pip: No such file or directory
```
the Can you have a try by replacing `./pip install opentuner` with `python -m pip install oprntuner`?

**Comment by Hecmay** [comment]
@zhangzhiru I cannot reproduce the error mentioned in this issue. There is nothing wrong in my installing DATuner revamp using the current Makefile.

I suggest @zhouyuan1119 to replace  `./pip install opentuner` with `./python -m pip install oprntuner` in line 45 of the `DATUNER_HOME/pkgs/python-opentuner/Makefile` and try again. If it works, we can update Makefile to avoid this issue.

**Comment by zhouyuan1119** [comment]
To make it clear, let me explain my environment settings when installation failed last time. For testing purposes I installed the master branch of DATuner on zhang-x1. When I tried to install the revamp branch, I forgot to unset the environment variables so I was using the `python` and `pip` that came with the master branch. The error occurred because the `pip` in the master branch messed up the installation. 

I just tried with the exact same setting, but changing the commands in the Makefile as @Hecmay suggested. Installation succeeded. I think we should update the makefile.
